Backend/routes/customerOrderRoutes.js

```
@@ -181,27 +181,29 @@ router.put(
       order.status = "Cancelled";
       await order.save();
 
-      const retailerOrder = await RetailerOrder.findOne({ shopId: mongoose.Types.ObjectId(order.shopId) });
+      const retailerOrder = await RetailerOrder.findOne({ shopId: order.shopId });
 
       if (retailerOrder) {
-        const orderTotal = parseFloat(order.total.replace("$", ""));
-        retailerOrder.total = `$${(
-          parseFloat(retailerOrder.total.replace("$", "")) - orderTotal
-        ).toFixed(2)}`;
+        const orderTotal = parseFloat(order.total?.replace("$", "") || 0);
```

Biome (1.9.4)

[error] Use Number.parseFloat instead of the equivalent global.

ES2015 moved some globals into the Number namespace for consistency.
Safe fix: Use Number.parseFloat instead.


(lint/style/useNumberNamespace)